aws/resource_aws_dx_hosted_private_virtual_interface_accepter.go

```diff
@@ -0,0 +1,160 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"virtual_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpn_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"dx_gateway_id"},
+			},
+			"dx_gateway_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"vpn_gateway_id"},
+			},
+			"tags": tagsSchema(),
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+	}
+}
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	vgwIdRaw, vgwOk := d.GetOk("vpn_gateway_id")
+	dxgwIdRaw, dxgwOk := d.GetOk("dx_gateway_id")
+	if vgwOk == dxgwOk {
+		return fmt.Errorf(
+			"One of ['vpn_gateway_id', 'dx_gateway_id'] must be set to create a Direct Connect private virtual interface accepter")
+	}
+
+	vifId := d.Get("virtual_interface_id").(string)
+	req := &directconnect.ConfirmPrivateVirtualInterfaceInput{
+		VirtualInterfaceId: aws.String(vifId),
+	}
+	if vgwOk && vgwIdRaw.(string) != "" {
+		req.VirtualGatewayId = aws.String(vgwIdRaw.(string))
+	}
+	if dxgwOk && dxgwIdRaw.(string) != "" {
+		req.DirectConnectGatewayId = aws.String(dxgwIdRaw.(string))
+	}
+
+	log.Printf("[DEBUG] Accepting Direct Connect hosted private virtual interface: %#v", req)
+	_, err := conn.ConfirmPrivateVirtualInterface(req)
+	if err != nil {
+		return fmt.Errorf("Error accepting Direct Connect hosted private virtual interface: %s", err.Error())
+	}
+
+	d.SetId(vifId)
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Service:   "directconnect",
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("dxvif/%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+
+	if err := dxHostedPrivateVirtualInterfaceAccepterWaitUntilAvailable(d, conn); err != nil {
+		return err
+	}
+
+	return resourceAwsDxHostedPrivateVirtualInterfaceAccepterUpdate(d, meta)
+}
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	vif, err := dxVirtualInterfaceRead(d.Id(), conn)
+	if err != nil {
+		return err
+	}
+	if vif == nil {
+		log.Printf("[WARN] Direct Connect virtual interface (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("virtual_interface_id", vif.VirtualInterfaceId)
+	d.Set("vpn_gateway_id", vif.VirtualGatewayId)
+	d.Set("dx_gateway_id", vif.DirectConnectGatewayId)
+	if err := getTagsDX(conn, d, d.Get("arn").(string)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsDxHostedPrivateVirtualInterfaceAccepterUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
This resource should not perform anything during update.

Reply:
Tags can be modified - see https://github.com/terraform-providers/terraform-provider-aws/pull/3255#discussion_r197569050.